futures-cpupool/src/lib.rs

```diff
@@ -258,8 +269,10 @@ impl<F: Future> Future for MySender<F, Result<F::Item, F::Error>> {
 
     fn poll(&mut self) -> Poll<(), ()> {
         if let Ok(Async::Ready(_)) = self.tx.as_mut().unwrap().poll_cancel() {
-            // Cancelled, bail out
-            return Ok(().into())
+            if !self.keep_running_flag.load(Ordering::SeqCst) {
```

Comment:
Thank!, my concern was the following case:
1. `MySender` checks `keep_running_flag`
2. `CpuFuture::forget()` called that sets `keep_running_flag = true` and drops `CpuFuture`
3. `MySender` checks `self.tx.as_mut().unwrap().poll_cancel()` and cancels the computation.

It's unlikely, but I guess it's still possible so..